src/dmd/expressionsem.d

```diff
@@ -3103,6 +3103,42 @@ private extern (C++) final class ExpressionSemanticVisitor : Visitor
         result = exp;
     }
 
+    override void visit(UDAItem e)
+    {
+        bool err = false;
+
+        if (e.type)
+        {
+            result = e;
+            return;
+        }
+
+        for (size_t i = 0; i < e.exps.dim; i++)
+        {
+            Expression exp = (*e.exps)[i];
+            exp = exp.expressionSemantic(sc);
+            if (!exp.type)
```

Comment:
`assert(exp.type)` in case you dont manage to reproduce the case.

Reply:
to be clear. If at some point someone one day compiles code that triggers the assert failure that will allow to spot a place were expsema returns null instead of errorexp.